sunpy/net/tests/test_fido.py

```diff
@@ -478,3 +478,18 @@ def test_client_fetch_wrong_type(mock_fetch):
 
     with pytest.raises(TypeError):
         Fido.fetch(qr)
+
+
+@pytest.mark.remote_data
+def test_slice_jsoc():
+    tstart = '2011/06/07 06:32:45'
+    tend = '2011/06/07 06:35:15'
+    res = Fido.search(a.Time(tstart, tend), a.jsoc.Series('hmi.M_45s'),
+                      a.jsoc.Notify('jsoc@cadair.com'))
+
+    with pytest.warns(SunpyUserWarning):
+        result = Fido.fetch(res[0, 0])
+        assert isinstance(result, Results)
+
```

Comment:
Do we not need to with blocks to make sure the second one raises?

Reply:
Sorry I didn't understand 😅

Reply:
I meant to say that you slice the results twice and make sure that pytest checks that there is a warning. But without a second with block, can you confirm that the second one has also raised its warning?

Reply:
Ohh sorry, I thought it would be duplication of code. So I believe I would have to add `with` block for every slicing.

Reply:
or we can check for the count of warnings raised